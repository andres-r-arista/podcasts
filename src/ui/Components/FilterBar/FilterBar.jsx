import React from "react";

import PropTypes from "prop-types";

import { Badge } from "../Badge/";

export const FilterBar = ({ podcastsAmount, onChange }) => {
  return (
    <div className="flex justify-end mr-4" data-testid="filter-bar">
      <Badge
        label={podcastsAmount.toString()}
        className="mr-2 bg-sky-800 text-white font-bold"
      />
      <input
        type="text"
        className="border text-sm px-2 py-1 w-64 rounded"
        onChange={onChange}
        placeholder="Filter podcasts..."
        data-testid="filter-bar-input"
      />
    </div>
  );
};

FilterBar.propTypes = {
  podcastsAmount: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};
