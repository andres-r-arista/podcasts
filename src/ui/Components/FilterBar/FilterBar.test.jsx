import { render, screen, fireEvent } from "@testing-library/react";

import { FilterBar } from "./FilterBar";

const onChange = jest.fn();

describe("Tests in <FilterBar />", () => {
  test("must match the snapshot", () => {
    const { container } = render(<FilterBar podcastsAmount={5} onChange={onChange} />);
    expect(container).toMatchSnapshot();
  });

  test("must call onChange function when changing the input", () => {
    render(<FilterBar podcastsAmount={5} onChange={onChange} />);
    expect(onChange.mock.calls).toHaveLength(0);
    const input= screen.getByTestId("filter-bar-input");
    fireEvent.change(input, { target: { value: 'Hello' } });
    expect(onChange.mock.calls).toHaveLength(1);
  })

});
