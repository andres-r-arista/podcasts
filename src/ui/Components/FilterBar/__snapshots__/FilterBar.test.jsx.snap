// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FilterBar /> must match the snapshot 1`] = `
<div>
  <div
    class="flex justify-end mr-4"
    data-testid="filter-bar"
  >
    <div
      class="px-1 rounded-lg self-center mr-2 bg-sky-800 text-white font-bold"
      data-testid="badge"
    >
      5
    </div>
    <input
      class="border text-sm px-2 py-1 w-64 rounded"
      data-testid="filter-bar-input"
      placeholder="Filter podcasts..."
      type="text"
    />
  </div>
</div>
`;
